resolver/resolver.cc

```diff
@@ -1927,40 +1959,65 @@ class ResolveSignaturesWalk {
         for (auto part : parts) {
             if (!current.exists()) {
                 // First iteration
-                if (part != "") {
-                    if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
-                        e.setHeader(
-                            "The string given to `{}` must be an absolute constant reference that starts with `{}`",
-                            method, "::");
+                if (!package) {
+                    if (part != "") {
+                        if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
+                            e.setHeader(
+                                "The string given to `{}` must be an absolute constant reference that starts with `{}`",
+                                method, "::");
+                        }
+                        return;
+                    }
+                    current = core::Symbols::root();
+                    continue;
+                } else if (package) {
+                    if (part == "") {
+                        if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
+                            e.setHeader("The string given to `{}` should not be an absolute constant reference if a "
+                                        "package name is also provided",
+                                        method);
+                        }
+                        return;
                     }
-                    return;
-                }
 
-                current = core::Symbols::root();
-            } else {
-                auto member = ctx.state.lookupNameConstant(part);
-                if (!member.exists()) {
-                    if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
-                        auto prettyCurrent =
-                            current == core::Symbols::root() ? "" : "::" + current.data(ctx)->show(ctx);
-                        auto pretty = fmt::format("{}::{}", prettyCurrent, part);
-                        e.setHeader("Unable to resolve constant `{}`", pretty);
+                    auto rootName = core::NameRef(ctx.state, package->value).lookupMangledPackageName(ctx.state);
+                    // if the mangled name doesn't exist, then this means probably there's no package named this
+                    if (!rootName.exists()) {
+                        if (auto e = ctx.beginError(*packageLoc, core::errors::Resolver::LazyResolve)) {
+                            e.setHeader("Unable to find package: `{}`", rootName.toString(ctx));
```

Comment:
Could you add a test that exercises this branch? (Unless I'm just missing it and it's there already.)